src/Pickles/Pickles/Runner.cs

```diff
@@ -70,9 +70,8 @@ private static void ApplyTestResultsToFeatures(IContainer container, IConfigurat
         {
             var testResults = container.Resolve<ITestResults>();
 
-            foreach (var node in features)
+            foreach (var featureTreeNode in features.Where(p => p.GetType() == typeof(FeatureNode)).Select(p => (FeatureNode)p))
             {
-                var featureTreeNode = node as FeatureNode;
```

Comment:
refactored to use [OfType](https://msdn.microsoft.com/query/dev14.query?appId=Dev14IDEF1&l=EN-US&k=k(System.Linq.Enumerable.OfType%60%601);k(TargetFrameworkMoniker-.NETFramework,Version%3Dv4.5);k(DevLang-csharp)&rd=true). Was copy pasting sample code from ImageNode earlier. But I didn't refactored those.